src/node/node.service.ts

```diff
@@ -36,6 +36,35 @@ export class NodeService {
     private readonly config: ConfigService,
   ) { }
 
+  private async signAndBroadcastSponsor(type: 18 | 19, recipient: string): Promise<any> {
+    const signResponse = await this.api.signTransaction({
+      version: 1,
+      type,
+      recipient,
+      fee: this.config.getSponsorFee(),
+    });
+
+    if (signResponse instanceof Error) {
+      throw signResponse;
+    }
+
+    const broadcastResponse = await this.api.broadcastTransaction(signResponse.data);
+
+    if (broadcastResponse instanceof Error) {
+      throw broadcastResponse;
+    }
+
+    return broadcastResponse.data;
+  }
+
+  async sponsorAccount(recipient: string): Promise<any> {
+    return this.signAndBroadcastSponsor(18, recipient);
+  }
+
+  async cancelSponsor(recipient: string): Promise<any> {
+    return this.signAndBroadcastSponsor(19, recipient);
+  }
+
```

Comment:
Add `isSponsored(recipient: string)`, which should check if the recipient is sponsored by our node.

Reply:
done